### ISSUE TITLE
When 'Enable payments via saved cards' setting is disabled, Card fields do not load on Checkout

### ISSUE THREAD

**Issue by jessLundie** [Automattic/woocommerce-payments#3261]
### Describe the bug

Credit card fields do not load on Checkout page when `Enable payments via saved cards` setting is disabled.

This does not appear to impact Block checkout, but does impact checkout pages built through the `[woocommerce-checkout]` shortcode.

4417380-zen

### To Reproduce

1. Go to **Payments > Settings**
2. Disable `Enable payments via saved cards`
3. Save
4. Put a non-subscription product in the Cart and go to Checkout
5. See that credit card fields do not load

### Actual behavior
Credit card fields do not load on Checkout page when `Enable payments via saved cards` setting is disabled.

### Screenshots

![image](https://user-images.githubusercontent.com/39742956/139557362-40cc58e5-0b75-4485-a849-eec57cfd8c6f.png)

### Expected behavior

Card fields should load on Checkout even if `Enable payments via saved cards` setting is disabled. Customers should be able to check out with non-subscription products. An appropriate error message should be displayed for customers who try to check out with subscription products (or subscriptions should be set as Manual renewal). 

### Additional context

Site Status Report

<details> 

```
`
### WordPress Environment ###

WordPress address (URL): https://wcpayhappiness.mystagingwebsite.com
Site address (URL): https://wcpayhappiness.mystagingwebsite.com
WC Version: 5.8.0
REST API Version: ✔ 5.8.0
WC Blocks Version: ✔ 6.2.0
Action Scheduler Version: ✔ 3.3.0
WC Admin Version: ✔ 2.7.2
Log Directory Writable: ✔
WP Version: 5.8.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.4.25
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.78.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.31-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.8.0
WC Database Prefix: wp_
Total Database Size: 5.40MB
Database Data Size: 3.11MB
Database Index Size: 2.29MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.20MB + Index: 0.22MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.05MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.05MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.23MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.59MB + Engine InnoDB
wp_posts: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.13MB + Index: 0.14MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 18
global_product_addon: 1
page: 9
post: 1
product: 12
revision: 3
shop_coupon: 3
shop_order: 196
shop_order_refund: 18
yay-currency-manage: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (6) ###

Akismet Anti-Spam: by Automattic – 4.2.1
Robot Ninja Helper: by Prospress Inc. – 1.13.1
WooCommerce Blocks: by Automattic – 6.2.0
WooCommerce Payments: by Automattic – 3.2.2
WooCommerce Subscriptions: by WooCommerce – 3.1.6
WooCommerce: by Automattic – 5.8.0

### Inactive Plugins (3) ###

Jetpack: by Automattic – 10.3-beta
WooCommerce Beta Tester: by WooCommerce – 2.0.4
WooCommerce Product Add-ons: by WooCommerce – 4.1.0

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #282 - /checkout-blocks/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.9.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://wcpayhappiness.mystagingwebsite.com
Subscription Statuses: –
WooCommerce Account Connected: ❌ No
Report Cache Enabled: ✔ Yes
Cache Update Failures: ✔ 0 failure

### Store Setup ###

Country / State: United States (US) — California

### Payment Gateway Support ###

WooCommerce Payments: products
refunds
subscriptions
subscription_cancellation
subscription_suspension
subscription_reactivation
subscription_amount_changes
subscription_date_changes
subscription_payment_method_change
subscription_payment_method_change_customer
subscription_payment_method_change_admin
multiple_subscriptions

Cash on delivery: products

### Action Scheduler ###

Complete: 61
Oldest: 2021-09-30 18:21:28 +0000
Newest: 2021-10-30 19:55:56 +0000

Pending: 2
Oldest: 2021-10-30 20:54:44 +0000
Newest: 2021-10-31 18:54:45 +0000


### Status report information ###

Generated at: 2021-10-30 20:17:23 +00:00
`
```
</details>

**Comment by jessLundie** [comment]
Also reported in 2486960-hc.

**Comment by grig-23** [comment]
4414019-zen

**Comment by grig-23** [comment]
4413662-zen

**Comment by haszari** [comment]
Reproduced on my [test site](https://ephemeral-haszari-20211001.atomicsites.blog) with subscription products also. 

@brucealdridge & @Automattic/helix is investigating (since we're online right now) - all welcome to assist :)

**Comment by aprea** [comment]
Fixed in https://github.com/Automattic/woocommerce-payments/pull/3264, this will be released in WooCommerce Payments 3.2.3
